### DISCUSSION TITLE
failed to start v2ray-core: not support "redirect" mode of transparent proxy

### DISCUSSION THREAD

**Discussion by ElectroBandit** [v2rayA/v2rayA#1498] [Q&A]
I'm trying to use v2rayA on an ASUS router with Merlin firmware and Entware. v2rayA installed successfully (with v2ray-core), and I was able to add my server, but I'm encountering the error:

`failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables -w 2 -t nat -N TP_OUT Bad argument `2' Try `iptables -h' or 'iptables --help' for more information. : exit status 2`

Is there any way to fix this?

**Comment by iopq** [comment]
try tproxy

**Reply by r0otx** [reply]
tproxy too shown this error

`failed to start v2ray-core: not support "tproxy" mode of transparent proxy: ExecCommands: iptables -w 2 -t mangle -N TP_OUT Bad argument `2' Try `iptables -h' or 'iptables --help' for more information. : exit status 2
`

**Comment by iopq** [comment]
oh, -w 2 is to wait two seconds

maybe your iptables version is different?

**Reply by r0otx** [reply]
> ой, -w 2 — это подождать две секунды
> 
> может быть у вас другая версия iptables?

I'm use package from entware repository. Run command opkg install iptables and restart router, may be need install custom version?

**Reply by giceli** [reply]
I also encounted this problem in my ASUS router, iptables do not support -w option, is there any methos to remove -w option in ExecCommands,
![image](https://github.com/user-attachments/assets/b4054a1f-7c8c-4959-a466-c939580ae2ab)

**Reply by iopq** [reply]
> > ой, -w 2 — это подождать две секунды
> > может быть у вас другая версия iptables?
> 
> I'm use package from entware repository. Run command opkg install iptables and restart router, may be need install custom version?

 I switched over to daed, you can try it
https://github.com/daeuniverse/daed
